source/cipher.c

```diff
@@ -88,6 +88,15 @@ static enum aws_cryptosdk_sha_version aws_cryptosdk_which_sha(enum aws_cryptosdk
     }
 }
 
+bool aws_cryptosdk_alg_properties_is_valid(const struct aws_cryptosdk_alg_properties *const alg_props) {
```

Comment:
The comment on `ensure_is_allocated` also belongs here, to explain the check that we're doing that this is a singleton.